Add Emogrifier migration guide for 5.4 (#103)
WoltLab/woltlab.github.io: pages/migration/wsc-53/migration_wsc-53_libraries.md

A detailed [Guzzle migration guide](https://github.com/guzzle/guzzle/blob/master/UPGRADING.md#60-to-70) can be found in the Guzzle documentation.

The explicit `sink` that was recommended in the [migration guide for WSC 5.2](migration_wsc-52_libraries.html#guzzle) can now be removed, as [the Guzzle issue #2735](https://github.com/guzzle/guzzle/issues/2735) was fixed in Guzzle 7.

## Emogrifier / CSS Inliner

The Emogrifier library was updated from version 2.2 to 5.0.

> Version 3.0 deprecated the `Emogrifier` class in favor of the `CssInliner` class that was added in 2.1.

Reviewer: ```suggestion
Version 3.0 deprecated the `Emogrifier` class in favor of the `CssInliner` class, which was added in 2.1.
```

> With the updated Emogrifier library the `CssInliner` class must be used instead.

Reviewer: ```suggestion
With the updated Emogrifier library, the `CssInliner` class must be used instead.
```

> The `Emogrifier` class was subsequently removed in version 4.0.

Reviewer: May-be just state to use `CssInliner` instead of `Emogrifier`. The exact history of when the changes appear to irrelevant here.

Author: I considered that important, because the Emogrifier class was removed without deprecating it within our software first, because they released so quickly that we missed several versions.

Reviewer: I understand but the details of when it happened is irrelevant for our documentation, in my opinion. Just stating that the `Emogrifier` class was removed and that the `CssInliner` class has to be used instead should suffice.

Author: I've rephrased that paragraph, PTAL.

> No compatibility layer was added for the `Emogrifier` class, as the Emogrifier library's purpose was for use within the email subsystem of WoltLab Suite.

Reviewer: ```suggestion
No compatibility layer was added for the `Emogrifier` class, as the Emogrifier library's (only?) purpose was to be used within the email subsystem of WoltLab Suite.
```

`(only?)` -> may-be add `only`?

Author: ```suggestion
No compatibility layer was added for the `Emogrifier` class, as the Emogrifier library's purpose was to be used within the email subsystem of WoltLab Suite.
```
We did not communicate that it was the “only” purpose, but it certainly was not meant to be used standalone without exactly knowing what you are doing. I'll leave out the “only”.

> In case you use Emogrifier directly within your own code you will need to adjust the usage.

Reviewer: ```suggestion
In case you use Emogrifier directly within your own code, you will need to adjust the usage.
```

Refer to the [Emogrifier CHANGELOG](https://github.com/MyIntervals/emogrifier/blob/v5.0.0/CHANGELOG.md) and [WoltLab/WCF #3738](https://github.com/WoltLab/WCF/pull/3738) if you need help making the necessary adjustments.

If you only use Emogrifier indirectly by sending HTML mail via the email subsystem then you might notice unexpected visual changes due to the improved CSS support.

> Double check your CSS declarations and specifically the specificity of your selectors in these cases.

Reviewer: ```suggestion
Double check your CSS declarations and particularly the specificity of your selectors in these cases.
```

`specifically the specificity` is be too specific ;)
